app/realtime.php

```diff
@@ -380,8 +380,10 @@ function getCache(): Cache
                         $user = $database->getDocument('users', $userId);
 
                         $roles = Auth::getRoles($user);
+                        $channels = $realtime->connections[$connection]['channels'];
 
-                        $realtime->subscribe($projectId, $connection, $roles, $realtime->connections[$connection]['channels']);
+                        $realtime->unsubscribe($connection);
+                        $realtime->subscribe($projectId, $connection, $roles, $channels);
```

Comment:
would it makes sense to call `unsubscribe($connection)` inside of `subscribe($projectId, $connection, $roles, $channels)` that way we can prevent such issues in the first place

Reply:
1. I'm wary about unsubscribing for every subscribe when it's not needed as it would add extra processing
2. The `subscribe()` method might get messy trying to handle old and new channels:

```diff
diff --git a/src/Appwrite/Messaging/Adapter/Realtime.php b/src/Appwrite/Messaging/Adapter/Realtime.php
index 55d8db292..fe89061d2 100644
--- a/src/Appwrite/Messaging/Adapter/Realtime.php
+++ b/src/Appwrite/Messaging/Adapter/Realtime.php
@@ -45,10 +45,14 @@ class Realtime extends Adapter
      * @param array $channels
      * @return void
      */
-    public function subscribe(string $projectId, mixed $identifier, array $roles, array $channels): void
+    public function subscribe(string $projectId, mixed $identifier, array $roles, array $newChannels): void
     {
         if (!isset($this->subscriptions[$projectId])) { // Init Project
             $this->subscriptions[$projectId] = [];
+        } else {
+            $oldChannels = $this->connections[$identifier]['channels'] ?? [];
+            $channels = array_unique(array_merge($oldChannels, $newChannels));
+            $this->unsubscribe($identifier);
         }
 
         foreach ($roles as $role) {
```

Reply:
I mean we can keep the `unsubscribe` method - but call it basically. If not calling unsubscribe before subscribing again, I dont think it will hurt.

Reply:
I wasn't suggesting removing the `unsubscribe` method. I was worried about adding extra processing of iterating over every role/channel when it's not needed considering realtime needs to be super fast.

Also, it's messy to handle both old and new channels in the subscribe method. Please see the diff above.